lib/cocoapods-binary-cache/pod-rome/xcodebuild_command.rb

```diff
@@ -166,11 +218,42 @@ def collect_output(target, paths)
     end
 
     def target_products_dir_of(target, sdk)
-      "#{build_dir}/#{configuration}-#{sdk}/#{target.name}"
+      if should_archive?
+        "#{build_dir}/#{configuration}-#{sdk}/#{target.name}.xcarchive"
+      else
+        "#{build_dir}/#{configuration}-#{sdk}/#{target.name}"
+      end
+    end
+
+    # only produced when archive xcframework
+    def framework_product_path_of(target, sdk)
+      "#{target_products_dir_of(target, sdk)}/Products"
     end
 
     def framework_path_of(target, sdk)
-      "#{target_products_dir_of(target, sdk)}/#{target.product_module_name}.framework"
+      if should_archive?
+        "#{target_products_dir_of(target, sdk)}/Products/Library/Frameworks/#{target.product_module_name}.framework"
+      else
+        "#{target_products_dir_of(target, sdk)}/#{target.product_module_name}.framework"
+      end
+    end
+
+    def dsym_path_of(target, sdk)
+      if should_archive?
+        "#{target_products_dir_of(target, sdk)}/dSYMs"
+      else
+        "#{target_products_dir_of(target, sdk)}/#{target.product_module_name}.framework.dSYM"
+      end
+    end
+
+    def bcsymbolmap_path_of(target, sdk)
+      "#{target_products_dir_of(target, sdk)}/BCSymbolMaps"
+    end
+
+    def should_archive?
+      # Use archive instead of build only when we have to generate bcsymbolmap or dSYM for xcframeworks,
+      # because archiving must be done on each scheme, hence it takes much more time, while building can be done simultaneously on all targets
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/grab/cocoapods-binary-cache/blob/b144f30fb41bd34e49fceb072710e5f2c3a3e710/lib/cocoapods-binary-cache/pod-rome/xcodebuild_command.rb#L255"></span>Layout/LineLength: Line is too long. [142/120]</td>
    </tr>
  </tbody>
</table>